sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -571,14 +569,36 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
 
       std::unique_ptr<HttpPolicy> Clone() const override
       {
-        return std::make_unique<BearerTokenAuthenticationPolicy>(
-            m_credential, m_tokenRequestContext);
+        return std::make_unique<BearerTokenAuthenticationPolicy>(*this);
       }
 
+      BearerTokenAuthenticationPolicy(BearerTokenAuthenticationPolicy const& other)
```

Comment:
Should these be virtual to prevent tearing in derived classes?

Reply:
Thank you - of course!
Good catch.
I put it that way while fighting with large amount of compiler errors with a thought to make it compile and deal with it later, but forgot to get to it.

I fixed it - now it is protected, and while we can't use that with `std::make_unique`, we are initializing it explicitly for the same effect.